tasks/eva_2124/correct_assembly_for_study.py

```diff
@@ -0,0 +1,112 @@
+        synonym_dictionaries = load_synonyms_for_assembly(assembly_accession)
+
+        insert_statements = []
+        drop_statements = []
+        record_checked = 0
+        already_genbanks = 0
+        total_inserted = 0
+        total_dropped = 0
+        for variant in cursor:
+            # Ensure that the variant we are changing has the expected SHA1
+            original_id = get_SHA1(variant)
+            assert variant['_id'] == original_id, "Original id is different from the one calculated %s != %s" % (variant['_id'], original_id)
+            genbank, was_already_genbank = get_genbank(synonym_dictionaries, variant['contig'])
+            if was_already_genbank:
+                already_genbanks += 1
+            else:
+                variant['contig'] = genbank
+                variant['_id'] = get_SHA1(variant)
+                insert_statements.append(pymongo.InsertOne(variant))
+                drop_statements.append(pymongo.DeleteOne({'_id': original_id}))
+            record_checked += 1
+            if len(insert_statements) >= chunk_size:
+                total_inserted, total_dropped = execute_bulk(drop_statements, insert_statements, sve_collection,
+                                                             total_dropped, total_inserted)
+
+        if len(insert_statements) > 0:
+            total_inserted, total_dropped = execute_bulk(drop_statements, insert_statements, sve_collection,
+                                                         total_dropped, total_inserted)
+        print('Retrieved %s documents and checked matching Sha1 hash' % record_checked)
+        print('{} of those documents had already a genbank contig. If the projects were all affected, '
+              'that number should be 0, but even if it is not, there is nothing else to fix'.format(already_genbanks))
+
+        print('There was %s new documents inserted' % total_inserted)
+        print('There was %s old documents dropped' % total_dropped)
+        return total_inserted
+
+
+def execute_bulk(drop_statements, insert_statements, sve_collection, total_dropped, total_inserted):
+    result_insert = sve_collection.bulk_write(requests=insert_statements, ordered=False)
+    total_inserted += result_insert.inserted_count
+    result_drop = sve_collection.bulk_write(requests=drop_statements, ordered=False)
+    total_dropped += result_drop.deleted_count
+    insert_statements.clear()
+    drop_statements.clear()
+    return total_inserted, total_dropped
+
+
+def main():
+    argparse = ArgumentParser()
+    argparse.add_argument('--mongo_user', help='user to connect to mongodb', required=True)
+    argparse.add_argument('--mongo_password', help='password to connect to mongodb', required=True)
+    argparse.add_argument('--mongo_host', help='host to connect to mongodb', required=True)
+    argparse.add_argument('--studies', help='The studies in the assembly to correct', required=True)
```

Comment:
You can allow space separated list of studies with the parameters `nargs='+'`. `args.studies` will then return a list.